src/pudl/transform/eiaaeo.py

```diff
@@ -0,0 +1,324 @@
+    )
+
+    assert (
+        sanitized.region_series == sanitized.region_category
+    ).all(), "Series and category must agree on region!"
+    assert (
+        sanitized.case_series == sanitized.model_case_eiaaeo
+    ).all(), "Series and taxonomy must agree on case!"
+
+    # TODO 2024-04-20: one day we'll include report year in the extract step -
+    # at that point we'll have to compare that with the report year extracted
+    # from the series name.
+    return sanitized.drop(columns=["region_series", "case_series"]).rename(
+        columns={"region_category": "region", "report_year_series": "report_year"}
+    )
+
+
+def unstack(df: pd.DataFrame, eventual_pk: list[str]):
+    """Unstack the values by the various variable names provided.
+
+    2024-04-27: NB it's important that `projection_year` is the last level of
+    index, so that when we sort and diff() we are actually subtracting year
+    over year.
+    """
+    assert (
+        eventual_pk[-1] == "projection_year"
+    ), "Projection year must be last level of index!"
+    unstacked = (
+        df.set_index(eventual_pk + ["variable_name"])
+        .unstack(level="variable_name")
+        .sort_index()
+    )
+    unstacked.columns = unstacked.columns.droplevel()
+    unstacked.columns.name = None
+    return unstacked
+
+
+@asset(io_manager_key="pudl_io_manager")
+def core_eiaaeo__yearly_projected_generation_in_electric_sector_by_technology(
+    raw_eiaaeo__electric_power_projections_regional,
+):
+    """Projected net summer generation capacity and additions/retirements."""
+
+    def _drop_aggregated_generation_data(df: pd.DataFrame) -> pd.DataFrame:
+        """These "aggregate" data series aren't split out by technology.
+
+        We'll report them elsewhere - but for this resource we filter them out.
+        """
+        generation_mask = df.variable_name == "generation"
+        is_aggregated_mask = df.dimension.isin(
+            {
+                "total_generation",
+                "sales_to_customers",
+                "generation_for_own_use",
+            }
+        )
+        return df.loc[~(generation_mask & is_aggregated_mask)]
+
+    def _collect_totals(df: pd.DataFrame) -> pd.DataFrame:
+        """Various total columns are named differently - they don't have to be."""
+        return df.assign(
+            dimension=df.dimension.str.replace(r"^total.*$", "total", regex=True)
+        )
+
+    def _decumulate(df: pd.DataFrame, decumulate_columns: list[str]) -> pd.DataFrame:
+        """De-accumulate the cumulative columns so we get yearly values.
+
+        Cumulative values are less useful because they require the user to know
+        what year the accumulation starts.
+        """
+        new_columns = {}
+        for decumulate_column in decumulate_columns:
+            cumulative_colname = f"cumulative_{decumulate_column}"
+            column = df[cumulative_colname]
+            new_columns[decumulate_column] = column.diff().fillna(column)
```

Comment:
nice! i really like the accumulation of a dict for the assign.

okay so this assumes/expects that the index is sorted and have that date col as the last col in the mutli-index? this might be personal preference, but I think that assertion in `unstack` might make more sense at the start of this function with a direct reference to the fact that the index is set in `unstack`. I just think without that the assumptions about what the data looks like when it comes into this function feel pretty important and unstated.

Reply:
Yeah, that makes a lot of sense. Moving